filebeat/harvester/log.go

```diff
@@ -148,11 +160,12 @@ func (h *Harvester) shouldExportLine(line string) bool {
 // the file system is scanned
 func (h *Harvester) openFile() error {
 
-	f, err := file.ReadOpen(h.path)
+	f, err := file.ReadOpen(h.state.Source)
```

Comment:
After doing `file.ReadOpen` a number of checks that might return an error are executed. Question: in case of openFile returning an error, who is closing the file handle?